hazelcast-jet-core/src/test/java/com/hazelcast/jet/core/MultigraphTest.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.core;
+
+import com.hazelcast.jet.JetInstance;
+import com.hazelcast.jet.core.TestProcessors.ListSource;
+import com.hazelcast.jet.core.processor.SinkProcessors;
+import org.junit.Test;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static com.hazelcast.jet.core.Edge.from;
+import static com.hazelcast.jet.function.Functions.wholeItem;
+import static java.util.function.Function.identity;
+import static java.util.stream.Collectors.toMap;
+import static org.junit.Assert.assertEquals;
+
+public class MultigraphTest extends JetTestSupport {
+
+    @Test
+    public void test() {
+        DAG dag = new DAG();
+        List<Integer> input = IntStream.range(0, 200_000).boxed().collect(Collectors.toList());
+        Vertex source = dag.newVertex("source", ListSource.supplier(input));
```

Comment:
test could rewritten to use the new test sources

Reply:
The new test sources are for pipeline API. I would have to use:
```java
((BatchSourceTransform) TestSources.items(input)).metaSupplier
```

Reply:
I think we can create processor version of those sources and get rid of ListSource